packages/core/content-manager/server/controllers/relations.js

```diff
@@ -104,7 +104,7 @@ module.exports = {
       const alias = subQuery.getAlias();
 
       const knexSubQuery = subQuery
-        .where({ id: entityId })
+        .where({ id: entityId, [`${alias}.id`]: { $notNull: true } })
```

Comment:
what is the alias?
From the url in the tests: `/content-manager/relations/api::shop.shop/${fieldName}`,
Is it referencing the fieldName ?

Reply:
Just to check if I understand the query.
Before the query was only looking for the entityID, but did not check if there were no relations.

What did happen before?

Reply:
The variable is defined above at line 104. It's actually as sql alias for a the target field table (here it's `t1`).
The full query that is done is (an example with countries having categories):
```sql
SELECT
  `t0`.`id`,
  `t0`.`name`,
  `t0`.`published_at`
FROM
  `categories` AS `t0`
WHERE ((`t0`.`locale` = 'en'
    and(`t0`.`id` NOT in(SELECT DISTINCT
          `t1`.`id` FROM `addresses` AS `t0`
        LEFT JOIN `addresses_categories_links` AS `t2` ON `t0`.`id` = `t2`.`address_id`
        LEFT JOIN `categories` AS `t1` ON `t2`.`category_id` = `t1`.`id`
      WHERE (`t0`.`id` = '3')))))
ORDER BY
  `t0`.`name` ASC
LIMIT 10;
```

The subquery done is the in the `in` :
```sql
SELECT DISTINCT
  `t1`.`id`
FROM
  `addresses` AS `t0`
  LEFT JOIN `addresses_categories_links` AS `t2` ON `t0`.`id` = `t2`.`address_id`
  LEFT JOIN `categories` AS `t1` ON `t2`.`category_id` = `t1`.`id`
WHERE (`t0`.`id` = '3');
```

The problem was that the result of the subquery was `NULL`. So it was doing `NOT IN (NULL)` which is not supported by SQL. The idea here is that the query doesn't return anything to have `NOT IN ()`.
It was returning NULL because it is doing left joins, so even if there are no categories, there is at least one row for the address (see screenshot).
![Capture d’écran 2022-09-08 à 11 31 22](https://user-images.githubusercontent.com/11708220/189087971-a6f2e205-24c0-4774-8455-f5fa155f7190.png)

Reply:
Oh! understood.
Thank you for the detailed explanation 🤗

Reply:
With pleasure 😊